tests/manage/configuration/migrations/test_migrations.py

```diff
@@ -25,108 +26,70 @@ class CircularMigration(Initial):
-        self.assertEqual(config_migrated, config_reference)
-
-    def test_4_5_migration(self):
-        """Test the step between config versions 4 and 5."""
-        config_initial = self.load_config_sample('input/4.json')
-        config_reference = self.load_config_sample('reference/5.json')
-        config_migrated = upgrade_config(config_initial, 5)
-        self.assertEqual(config_migrated, config_reference)
-
-    def test_5_6_migration(self):
-        """Test the step between config versions 5 and 6."""
-        config_initial = self.load_config_sample('input/5.json')
-        config_reference = self.load_config_sample('reference/6.json')
-        config_migrated = upgrade_config(config_initial, 6)
-        self.assertEqual(config_migrated, config_reference)
-
-    def test_6_5_migration(self):
-        """Test the step between config versions 6 and 5."""
-        config_initial = self.load_config_sample('reference/6.json')
-        config_reference = self.load_config_sample('input/5.json')
-        config_migrated = downgrade_config(config_initial, 5)
-        self.assertEqual(config_migrated, config_reference)
-
-    def test_6_7_migration(self):
-        """Test the step between config versions 6 and 7."""
-        config_initial = self.load_config_sample('input/6.json')
-        config_reference = self.load_config_sample('reference/7.json')
-        config_migrated = upgrade_config(config_initial, 7)
-        self.assertEqual(config_migrated, config_reference)
+    return _factory
+
+
+def test_upgrade_path_fail(load_config_sample):
+    """Test failure when no upgrade path is available."""
+    config_initial = load_config_sample('reference/5.json')
+    # target lower than initial
+    with pytest.raises(ConfigurationError):
+        upgrade_config(copy.deepcopy(config_initial), 1)
+    # no migration available
+    with pytest.raises(ConfigurationError):
+        upgrade_config(copy.deepcopy(config_initial), 100)
+    # circular dependency
+    with pytest.raises(ConfigurationError):
+        upgrade_config(config_initial, 6, migrations=[CircularMigration])
+
+
+def test_downgrade_path_fail(load_config_sample):
+    """Test failure when no downgrade path is available."""
+    config_initial = load_config_sample('reference/5.json')
+    # target higher than initial
+    with pytest.raises(ConfigurationError):
+        downgrade_config(copy.deepcopy(config_initial), 6)
+    # no migration available
+    with pytest.raises(ConfigurationError):
+        downgrade_config(copy.deepcopy(config_initial), -1)
+    # circular dependency
+    with pytest.raises(ConfigurationError):
+        downgrade_config(config_initial, 4, migrations=[CircularMigration])
+
+
+def test_migrate_full(load_config_sample, monkeypatch):
+    """Test the full config migration."""
+    config_initial = load_config_sample('input/0.json')
+    monkeypatch.setattr(uuid, 'uuid4', lambda: uuid.UUID(hex='0' * 32))
```

Comment:
Because the migration will generate a random UUID, but we are checking against a static reference file that has `0 * 32` as UUID. If we don't force the same UUID, the comparison and so test will fail

Reply:
Mmm I see, thanks. I'm wondering if it makes sense to add a comment for this since we are not technically monkeypatching to isolate the module or to verify that the `uuid` lib is called, but just as a side-effect of how we do the tests, i.e. wanting to do direct reference file comparison (since if we were doing it manually migration by migration we would only to check that the 'PROFILE_UUID' keyword was added and not its content). Not sure how important it is to clarify this distinction though, so up to you, I'll approve either way.